tests/agent_server/test_resource_handler.py

```diff
@@ -0,0 +1,63 @@
+"""
+    Copyright 2019 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+import base64
+import typing
+from typing import TypeVar
+
+import pytest
+
+from inmanta.agent.handler import ResourceHandler
+from inmanta.protocol import common
+from test_protocol import make_random_file
+
+T = TypeVar("T")
+
+
+class MockGetFileResourceHandler(ResourceHandler):
+    def __init__(self, return_code, content):
+        self.return_code = return_code
+        self.content = content
+        pass
+
+    def run_sync(self, func: typing.Callable[[], T]) -> T:
+        content = b""
```

Comment:
1. I would add a comment as to why you mock run_sync and not get_file 
2. I would add an assert of the form assert func.`__func__` == ResourceHandler.get_file, to ensure clean failure if the getfile method is modified